openedx/core/djangoapps/programs/management/commands/backpopulate_program_credentials.py

```diff
@@ -0,0 +1,112 @@
+logger = logging.getLogger(__name__)
+RunMode = namedtuple('RunMode', ['course_key', 'mode_slug'])
+
+
+class Command(BaseCommand):
+    """Management command for backpopulating missing program credentials.
+
+    The command's goal is to pass a narrow subset of usernames to an idempotent
+    Celery task for further (parallelized) processing.
+    """
+    help = 'Backpopulate missing program credentials.'
+    client = None
+    run_modes = None
+    usernames = None
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-c', '--commit',
+            action='store_true',
+            dest='commit',
+            default=False,
+            help='Submit tasks for processing.'
+        )
+
+    def handle(self, *args, **options):
+        programs_config = ProgramsApiConfig.current()
+        self.client = Client.objects.get(name=programs_config.OAUTH2_CLIENT_NAME)
+
+        if self.client.user is None:
+            msg = (
+                'No user is associated with the %s OAuth2 client. No tasks have been enqueued. '
+                'Associate a user with the client and try again.'
+            ).format(programs_config.OAUTH2_CLIENT_NAME)
+
+            raise CommandError(msg)
+
+        self._get_run_modes()
+
+        logger.info('Looking for users who may be eligible for a program certificate.')
+
+        self._get_usernames()
+
+        if options.get('commit'):
+            logger.info('Enqueuing tasks for %d eligible users.', len(self.usernames))
+        else:
+            logger.info(
+                'Found %d eligible users. To enqueue tasks, pass the -c or --commit flags.',
+                len(self.usernames)
+            )
+            return
+
+        succeeded, failed = 0, 0
+        for username in self.usernames:
+            try:
+                award_program_certificates.delay(username)
+            except:  # pylint: disable=bare-except
+                failed += 1
+                logger.exception('Failed to enqueue program certification task for user [%s]', username)
+            else:
+                succeeded += 1
+                logger.info('Successfully enqueued task for user [%s]', username)
+
+        logger.info('Done. %d succeeded. %d failed.', succeeded, failed)
+
+    def _get_run_modes(self):
+        """Find all run modes which are part of a program."""
+        programs = get_programs(self.client.user)
+        self.run_modes = self._flatten(programs)
+
+    def _flatten(self, programs):
+        """Flatten program dicts into a list of unique run modes."""
+        run_modes = []
+        for program in programs:
+            for course_code in program['course_codes']:
+                for run in course_code['run_modes']:
+                    course_key = CourseKey.from_string(run['course_key'])
+                    run_mode = RunMode(course_key, run['mode_slug'])
+
+                    if run_mode not in run_modes:
+                        run_modes.append(run_mode)
```

Comment:
If the order of run modes isn't important (which it doesn't seem to be), you could just use a set.